api/routes/affiliations/post.js

```diff
@@ -0,0 +1,19 @@
+const { raw: knex } = require('../../db');
+const { loggedIn } = require('../../middleware/auth');
+
+module.exports = (app) => {
+  app.post('/states/:stateId/affiliations', loggedIn, async (request, response) => {
+    const userId = request.user.id;
+    const { stateId } = request.params;
```

Comment:
won't you need the role_id too?

Reply:
A `POST /states/{xx}/affiliations` request is a user requesting to be affiliated with a state. Should the user choose their role, or should the admin above them choose?